crates/rust-analyzer/src/reload.rs

```diff
@@ -505,7 +511,13 @@ impl SourceRootConfig {
     }
 }
 
-pub(crate) fn load_proc_macro(client: Option<&ProcMacroServer>, path: &AbsPath) -> Vec<ProcMacro> {
+/// Load the proc-macros for the given lib path, replacing all expanders whose names are in `dummy_replace`
+/// with an identity dummy expander.
+pub(crate) fn load_proc_macro(
+    client: Option<&ProcMacroServer>,
+    path: &AbsPath,
+    dummy_replace: &[Box<str>],
```

Comment:
I wonder if perhaps we should implement this on the higher level and treat such macros as inert attributes?

ah, I see you’ve considered that. I wonder if we can treat this as a custom tool attribute (#[rustfmt and such)

Reply:
tool attribute would be nice in theory, but since those are unstable and therefor require a feature flag I am doubtful that people would even bother using it

Reply:
I mean, we presumably have some infrastruture internally to track tool/intert attributes and say things like "`inline` does nothing". We can use that same infrastructure to implement this feature. Ie, this is about impl detail rather thatn about API. 

Basically, make `#[tokio::main]`  behave like `#[inline]` rather than `#[identity]`.